### DISCUSSION TITLE
How to change the name of the app that appears in the Windows taskbar and task manager?

### DISCUSSION THREAD

**Discussion by morphinapg** [AvaloniaUI/Avalonia#12075] [Q&A] [answered]
It's using the default [project name].Desktop on the taskbar and task manager. I have a custom title set in the Main Window XAML, and I also set the Name property in App.axaml, but it's still showing the default name in the taskbar and task manager. I tried looking around in the project properties, but the settings I tried changing didn't fix this either. What am I missing?

**Comment by timunie** [comment]
I believe this is not controlled by Avalonia but by .NET itself. See this thread for example: https://social.msdn.microsoft.com/Forums/vstudio/en-US/c2aeb03b-e86d-4f67-b3d1-6512b5c5a3a6/how-to-change-exe-name-cvery-urgent?forum=csharpgeneral

**Reply by morphinapg** [reply]
Okay so this is only the exe filename then? I figured the name shown there could be different than the exe name. Regardless, I had tried that, but the name I was using was giving me an error at build

```
MSB4018	The "GenerateDepsFile" task failed unexpectedly.
System.ArgumentException: An item with the same key has already been added.
```

If I change the Assembly name to literally anything else, it works, but not the name I want.

**Reply by timunie** [reply]
Don't know. Some file names maybe invalid due to not allowed char.

**Reply by morphinapg** [reply]
That doesn't seem to be it. The name I'm using is "Youtube Earnings Projections". If I add an extra space, it works just fine. Based on the error I would guess there's some kind of conflict with that name being used elsewhere, but I'm not sure exactly where it's conflicting.

**Reply by timunie** [reply]
Can you prepare a very minimal sample to reproduce?

**Reply by morphinapg** [reply]
https://drive.google.com/file/d/1O4cveV-FNYBPajaso-ysPmMGVFPpVauw/view?usp=sharing

I just created a new project, set startup project to .Desktop, and set Assembly name in the .Desktop project properties.

**Reply by timunie** [reply, marked as answer]
In your sample, both projects have the exact same Assembly name, so the compiler cannot decide which one to choose. 
![image](https://github.com/AvaloniaUI/Avalonia/assets/47110241/48a6041b-76ee-4fd2-90c6-f610adef0489)

In order to resolve this, I changed assembly name for base project like that: 

![image](https://github.com/AvaloniaUI/Avalonia/assets/47110241/2894911b-80d6-4b0c-b3e7-0505b2c01119)

**Reply by morphinapg** [reply]
Thank you, this worked
